stwcs/wcsutil/altwcs.py

```diff
@@ -763,6 +736,20 @@ def _parpasscheck(fobj, ext, wcskey, fromext=None, toext=None, reusekey=False):
     return True
 
 
+def _check_wcskey(wcskey, allow_none=False, wcskey_par_name='wcskey'):
+    if wcskey is None:
+        if not allow_none:
+            raise TypeError(f"Parameter '{wcskey_par_name}' must be a string.")
+        else:
+            return
+
+    if len(wcskey) > 1 or wcskey.strip() not in string.ascii_letters:
```

Comment:
In principle it should. I was debating this myself. The reason to allow empty strings `''` is because many existing functions have them as default values so they are clearly permitted and then not supporting them anylonger could create issues in other packages that (or users who) use `stwcs`. So, this is for backward compatibility.

Reply:
Clearly the code will not run with empty strings, so what matters is are they allowed in the interface. I don't see any empty strings in the default values of any of the functions

Reply:
See https://github.com/spacetelescope/stwcs/blob/0e985fb0dc9280b1aea6d5b26c42d42f8788507a/stwcs/wcsutil/altwcs.py#L532

Reply:
... and, if I change to strict equality and run unit tests, I get:
```
stwcs/tests/test_altwcs.py EEEEEEEEEEEE                                                           [ 24%]
stwcs/tests/test_astrometrydb.py EEE                                                              [ 30%]
stwcs/tests/test_headerlet.py EEEEEEEEEEEEEE                                                      [ 59%]
stwcs/tests/test_hstwcs.py .                                                                      [ 61%]
stwcs/tests/test_updatewcs.py EEEEEEEFFF....FFFFF                                                 [100%]
```

Reply:
I re-did unit tests and with strict `len(wcskey) != 1` all tests pass. It is probably OK to do this as it is very unlikely this function is used by anyone to run with default parameters.